system/Autoloader/FileLocator.php

```diff
@@ -309,6 +311,9 @@ public function listFiles(string $path): array
 
             $tempFiles = get_filenames($fullPath, true);
 
+            // Remove directories
+            $tempFiles = array_filter($tempFiles, static fn ($path) => strtolower(substr($path, -4)) === '.php');
```

Comment:
Why not use the `is_file` check?

Reply:
`is_file()` is slower. It needs file system access.

```php
            // Remove directories
            timer('rmdir');
            for ($i = 0; $i < 10000; $i++) {
                $tempFiles = array_filter($tempFiles, static fn ($path) => is_file($path));
//                $tempFiles = array_filter($tempFiles, static fn ($path) => strtolower(substr($path, -4)) === '.php');
            }
            timer('rmdir');
            echo timer()->getElapsedTime('rmdir');
            exit;
```

is_file(): 3.3436
substr(): 1.5712

Reply:
I think it is better that `get_filenames()` returns only files without directories.